src/audit/routes/system.py

```diff
@@ -1,10 +1,74 @@
 from fastapi import APIRouter, FastAPI, Request, Depends
+from typing import Union, get_args, get_origin
 
 from ..db import DataAccessLayer, get_data_access_layer
 
+from ..models import CreateLoginLogInput, CreatePresignedUrlLogInput
 
 router = APIRouter()
 
+# Increment versions on schema/model change.
+CURRENT_SCHEMA_VERSIONS = {
+    "login": 2,
+    "presigned_url": 1,
+}
+
+
+def _pretty_type(t) -> str:
+    """
+    Human-readable short type name
+
+    str                -> "str"
+    list[str]          -> "list"
+    Union[str, None]   -> "str?"
+    """
+    origin = get_origin(t)
+
+    # handle Optional types (Union types with None)
+    if origin is Union and type(None) in get_args(t):
+        # filter out NoneType and assume first remaining arg as base, or fallback to object
+        non_none_args = [a for a in get_args(t) if a is not type(None)]
+        base = non_none_args[0] if non_none_args else object
```

Comment:
instead of falling back to object, i think it could simply raise an error because it doesn't make sense